controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -611,27 +611,20 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
                 resultFuture.completeExceptionally(new WireCommandFailedException(new ConnectionFailedException(e),
                         request.getType(),
                         WireCommandFailedException.Reason.ConnectionFailed));
-            } else {
-                try {
-                    connection.send(request);
-                } catch (ConnectionFailedException cfe) {
-                    throw new WireCommandFailedException(cfe,
-                            request.getType(),
-                            WireCommandFailedException.Reason.ConnectionFailed);
-                } catch (Exception e2) {
-                    throw new RuntimeException(e2);
-                }
-            }
-        }).exceptionally(e -> {
-            Throwable cause = Exceptions.unwrap(e);
-            if (cause instanceof WireCommandFailedException) {
-                resultFuture.completeExceptionally(cause);
-            } else if (cause instanceof ConnectionFailedException) {
-                resultFuture.completeExceptionally(new WireCommandFailedException(cause, request.getType(), WireCommandFailedException.Reason.ConnectionFailed));
-            } else {
-                resultFuture.completeExceptionally(new RuntimeException(cause));
+            } else {                
+                connection.sendAsync(request, cfe -> {
+                    if (cfe != null) {
+                        Throwable cause = Exceptions.unwrap(cfe);
+                        if (cause instanceof WireCommandFailedException) {
```

Comment:
In my opinion we should never get a WireCommanFailedException, since cfe ie the exception received by netty when it does `writeAndFlush`.